test/search-display-controller.test.js

```diff
@@ -0,0 +1,121 @@
+import {querySelectorNotNull} from '../ext/js/dom/query-selector.js';
+
+const documentSearchDisplayControllerEnv = await setupDomTest('ext/search.html');
+
+const {window, teardown} = documentSearchDisplayControllerEnv;
+
+const {document} = window;
+
+let queryInputCount = 0;
+
+/**
+ * @param {?Element} element
+ * @returns {boolean}
+ */
+function isElementInput(element) {
+    if (element === null) { return false; }
+    switch (element.tagName.toLowerCase()) {
+        case 'input':
+        case 'textarea':
+        case 'button':
+        case 'select':
+            return true;
+    }
+    return element instanceof HTMLElement && !!element.isContentEditable;
+}
+
+/**
+ * @type {HTMLInputElement}
+ */
+const queryInput = querySelectorNotNull(document, '#search-textbox');
+
+/**
+ * @param {KeyboardEvent} e
+ */
+function onKeyDownCurrent(e) {
+    const {activeElement} = document;
+    if (
+        activeElement !== queryInput &&
+        !isElementInput(activeElement) &&
+        (!e.ctrlKey || e.key === 'Backspace') &&
+        !e.metaKey &&
+        !e.altKey &&
+        (e.key.length === 1 || e.key === 'Backspace') &&
+        e.key !== ' '
+    ) {
+        queryInput.focus({preventScroll: true});
+        queryInputCount += 1;
+    }
+}
+
+/**
+ * @param {KeyboardEvent} e
+ */
+function onKeyDownNew(e) {
+    const activeElement = document.activeElement;
+    const isInputField = isElementInput(activeElement);
+    const isAllowedKey = e.key.length === 1 || e.key === 'Backspace';
+    const isModifierKey = e.ctrlKey || e.metaKey || e.altKey;
+    const isSpaceKey = e.key === ' ';
+
+    if (!isInputField && !isModifierKey && isAllowedKey && !isSpaceKey) {
+        queryInput.focus({preventScroll: true});
+        queryInputCount += 1;
+    }
+}
+
+describe('Keyboard Event Handling', () => {
+    afterAll(() => teardown(global));
+
+    const keypressEvents = [
+        new KeyboardEvent('keydown', {key: 'a', ctrlKey: false, metaKey: false, altKey: false}),
+        new KeyboardEvent('keydown', {key: '', ctrlKey: true, metaKey: false, altKey: false}),
+        new KeyboardEvent('keydown', {key: '', ctrlKey: false, metaKey: true, altKey: false}),
+        new KeyboardEvent('keydown', {key: '', ctrlKey: false, metaKey: false, altKey: true}),
+        new KeyboardEvent('keydown', {key: ' ', ctrlKey: false, metaKey: false, altKey: false}),
+        new KeyboardEvent('keydown', {key: 'a', ctrlKey: true, metaKey: false, altKey: false}),
+        new KeyboardEvent('keydown', {key: 'a', ctrlKey: false, metaKey: true, altKey: false}),
+        new KeyboardEvent('keydown', {key: 'a', ctrlKey: false, metaKey: false, altKey: true}),
+        new KeyboardEvent('keydown', {key: 'Backspace'}),
+        new KeyboardEvent('keydown', {key: 'Backspace', ctrlKey: true, metaKey: false, altKey: false}),
+        new KeyboardEvent('keydown', {key: 'ArrowDown'})
+    ];
+
+    test('should test that onKeyDownCurrent function focuses input', () => {
```

Comment:
btw I feel like we should include one or two tests that test the negative condition.

For example, if `onKeyDown`'s logic is 
```js
function onKeyDown(e){
    if (true) {
        queryInput.focus({preventScroll: true});
    }
}
```

Then these tests would pass even though it's wrong